Pygame dependency fix

apal6981 opened the pull request:
I was bothered that pygame was being imported by default. I changed that it only gets loaded when the screen is being simulated. The keyboard broadcaster was the culprit. I changed it that the keyboard is only checked when the virtual screen is being used. Also stuff from v2 merge so lets confirm and pull that one before this one.

apal6981 committed:
cleaning up pygame dependencies

apal6981 committed:
cleaning up pygame dependencies

apal6981 committed:
changed dependency on pygame, only loads when screen is being simulated

apal6981 committed:
changed dependency on pygame, only loads when screen is being simulated

apal6981 committed:
formated with black

apal6981 committed:
formated with black

christopolise commented on controllers/__init__.py:
```diff
@@ -1,7 +1,7 @@
 
-def start_inputs(system_queue, demo_input_queue):
+def start_inputs(system_queue, demo_input_queue, simulated):
     """
```
If a new parameter is added, it should be updated in the docstring comment

christopolise commented on demos/sweep/main.py:
```diff
@@ -39,14 +39,15 @@ def stop(self):
 
         Args:
             x (int): x location of the line
             val (int): value of the line
```
Please update docstring param to match the signature of the function (i.e. val -> value)

christopolise reviewed (changes requested):
Looks great for the most part, however when I run 

```bash
python3 main.py -vvv simulator 
```

I get an error saying that I needed to provide a value for `simulated`

```
Traceback (most recent call last):
  File "/home/christopolise/dev/sss/main.py", line 104, in <module>
    cli()
  File "/home/christopolise/dev/sss/venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/christopolise/dev/sss/venv/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/christopolise/dev/sss/venv/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/christopolise/dev/sss/venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/christopolise/dev/sss/venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/christopolise/dev/sss/main.py", line 30, in run_simulator
    simulator.run()
  File "/home/christopolise/dev/sss/runners/simulator.py", line 366, in run
    sim.start()
  File "/home/christopolise/dev/sss/runners/simulator.py", line 310, in start
    handle_input = controllers.start_inputs(self.system_q, self.input_q)
TypeError: start_inputs() missing 1 required positional argument: 'simulated'
```

When checking the help messages, I find no mention of changes of command line args for running things in a simulated environment, so maybe include some of those if we are changing the way we run things?

philipbl commented:
@apal6981, can you rebase this PR so that the changes that were previously accepted don’t show up in this PR (physical_screen2.py, etc.)?

apal6981 committed:
fix docstrings

apal6981 committed:
moved RPi/linux specific dependencies into its own requirements, they were breaking windows

apal6981 committed:
fixed linux specific python path programming, now the simulator works on windows

apal6981 committed:
formatted

apal6981 committed:
isort

apal6981 committed:
black

apal6981 committed:
ran isort correctly??

apal6981 committed:
rearranged requirements.txt files

apal6981 commented:
I believe i have addressed the issues that needed to be addressed. I might have botched the rebase so somebody might want to look at that. I got a bit lost trying to do it on the commandline.

christopolise committed:
Merge branch 'dev' of github.com:NET-BYU/sss into pygameDependencyFix

christopolise committed:
fix black formatting
